### PR TITLE
[dualtor]Implement traffic distribution utility.

### LINKED ISSUE #2819: Traffic Distribution Utility
Method to verify that IP-in-IP tunnel traffic between standby ToR and T1 is equally distributed between all active links (need to account for one link being down)

### PR DESCRIPTION
**Pull request by bingwang-ms**
Signed-off-by: bingwang <bingwang@microsoft.com>

### Description of PR

Summary:
Fixes #2819 
This PR implements a utility for checking balance in IPinIP tunnel.
A script ```ip_in_ip_tunnel_test.py``` will be running on ptf and generate traffic to standby ToR. The packets should be forwarded to active ToR through IPinIP tunnel, and the script on ptf will capture and verify if the traffic distribution is balanced.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
This PR is to implements a utility for checking balance in IPinIP tunnel.

#### How did you do it?
Please refer code.

#### How did you verify/test it?
Verified on dualtor testbed with a demo script.
```
def test_ip_in_ip_tunnel(duthosts, tbinfo, ptfhost):
    ptf_portchannel_intfs = get_t1_active_ptf_ports(duthosts[1], tbinfo)
    server_ip = '192.168.0.2'
    active_tor_ip = '10.1.0.32'
    standby_tor_ip = '10.1.0.33'
    active_tor_mac = 'd4:af:f7:4d:a4:44'
    standby_tor_mac = 'd4:af:f7:4d:af:18'

    check_tunnel_balance(ptfhost, active_tor_mac, standby_tor_mac, active_tor_ip, standby_tor_ip, server_ip, ptf_portchannel_intfs)
```
The demo srcipt passed without error.

#### Any platform specific information?
No.

#### Supported testbed topology if it's a new test case?
Dualtor testbed specific.

### Documentation 

No.

### CODE REVIEW

File: ansible/roles/test/files/ptftests/ip_in_ip_tunnel_test.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+                            ip_ttl=64)
+        return pkt
+
+    def generate_expected_packet(self, inner_pkt):
+        """
+        Generate ip_in_ip packet for verifying.
+        """
+        inner_pkt = inner_pkt.copy()
+        inner_pkt.ttl = inner_pkt.ttl - 1
+        """
```
Comment (lolyu) [inline ansible/roles/test/files/ptftests/ip_in_ip_tunnel_test.py:97]: The comment here should be removed, right?

**Review by theasianpianist — APPROVED** [review]

File: ansible/roles/test/files/ptftests/ip_in_ip_tunnel_test.py (hunk 1 above)
Comment (bingwang-ms) [reply ansible/roles/test/files/ptftests/ip_in_ip_tunnel_test.py]: Thanks. Updated.

File: ansible/roles/test/files/ptftests/ip_in_ip_tunnel_test.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        return str(ip_address(begin) + random.randint(0, length))
+
+    def generate_packet_to_server(self, hash_key):
+        """
+        Generate a packet to server. The value of field in packet is filled with random value according to hash_key
+        """
+        base_src_mac = self.dataplane.get_mac(0, 0)
+        ip_src = self.random_ip(SRC_IP_RANGE[0], SRC_IP_RANGE[1]) if hash_key == 'src-ip' else SRC_IP_RANGE[0]
+        ip_dst = self.server_ip
+        sport = random.randint(1, 65535) if hash_key == 'src-port' else 1234
```
Comment (prsunny) [inline ansible/roles/test/files/ptftests/ip_in_ip_tunnel_test.py:73]: I'm not sure if this random generation will always provide consistent results. May be a rerun might pass. Could you check if the logic is similar to existing fib hash test?

Comment (bingwang-ms) [reply ansible/roles/test/files/ptftests/ip_in_ip_tunnel_test.py:73]: Yes. The logic is almost the same with fib hash test. Theoretically, if we generate enough packets, the distribution will be even enough.

**Pull request merged**